workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentDataGenerator.tt

```diff
@@ -24,8 +24,81 @@ namespace <#= qualifiedNamespace #>
         {
             public uint ComponentId => <#= componentDetails.ComponentId #>;
 
-            public BlittableBool DirtyBit { get; set; }
-<# foreach(var fieldDetails in fieldDetailsList) { #>
+            private BlittableBool isDirty;
+
+            // Bit masks for tracking which component properties were changed locally and need to be synced.
+            // Each byte tracks 8 component properties.
+<# for (var i = 0; i < (fieldDetailsList.Count / 8) + 1; i++) { #>
+            private byte dirtyBits<#= i #>;
+<# } #>
+
+            public bool IsDirty()
+            {
+                return isDirty;
```

Comment:
I'd recommend removing this flag, as it's redundant.
`return dirtyBits0 || dirtyBits1 || ...;` is just as comprehensible.

Reply:
Yeah, this "optimisation" was probably unnecessary. Changed to your suggestion.